src/System Application/App/Rest Client/src/HttpContentImpl.Codeunit.al

```diff
@@ -113,31 +198,52 @@ codeunit 2355 "Http Content Impl."
     begin
         Json := AsText();
         if Json <> '' then
-            ReturnValue.ReadFrom(AsText());
+            if not ReturnValue.ReadFrom(AsText()) then
+                ThrowInvalidJsonException();
+    end;
+
+    procedure AsJsonObject() ReturnValue: JsonObject
+    var
+        Json: Text;
+    begin
+        Json := AsText();
+        if Json <> '' then
+            if not ReturnValue.ReadFrom(AsText()) then
+                ThrowInvalidJsonException();
+    end;
```

Comment:
NIT: personal preference is to always reduce nesting where possible:

```suggestion
    procedure AsJsonObject() ReturnValue: JsonObject
    var
        Json: Text;
    begin
        Json := AsText();
        if Json == '' then
            exit;
        if not ReturnValue.ReadFrom(AsText()) then
            ThrowInvalidJsonException();
    end;
```

Doesn't really matter in such small functions, but it's good practice to make code more easily readable. But feel free to disregard here.

Reply:
Done. I also removed duplicate usage of AsText() while at it.